### PR TITLE
Extract float

### PR DESCRIPTION
**Pull request by kaiquekandykoga**
It is changing the way size option is manipulated in image_tag and video_tag allowing to pass float numbers.

Thank you very much.

### CODE REVIEW

**Review by tenderlove — COMMENTED** [review]
This seems like a pretty complicated way to get the job done.  Can we simplify it by modifying the regular expression?  Also, I didn't know browsers support floating point dimensions.  Can you link to more information?  [Documentation that I found](https://www.w3.org/MarkUp/html3/img.html) says that the width and height should be specified in pixels, and AFAIK you can't split pixels so floating point numbers seem odd.

**Comment by kaiquekandykoga** [comment]
Hi @tenderlove ,

Firstly I would like to thank you for your review.

I created an example to show the difference when using decimal values for images.

first


second_pic


third


fourth




https://codepen.io/kaiquekandykoga/pen/xxpbjPR

I am using Mozilla Firefox and openSUSE.

File: actionview/lib/action_view/helpers/asset_tag_helper.rb [hunk 1]
```diff
@@ -517,11 +517,26 @@ def resolve_image_source(source, skip_pipeline)
 
         def extract_dimensions(size)
           size = size.to_s
-          if /\A\d+x\d+\z/.match?(size)
```
[the remark this answers is not in the document]
Comment (kaiquekandykoga) [reply actionview/lib/action_view/helpers/asset_tag_helper.rb:520]: I was avoiding having to work with complex regexp and I made it even harder to read.

Changed back to use only regexp. Thank you.

**Comment by tenderlove** [comment]
Cool.  Seems good to me!  Thanks for making this demo!

**Pull request merged**